spec/lib/retention_instructions_spec.rb

```diff
@@ -0,0 +1,184 @@
+
+# rubocop:disable RSpec/DescribeClass
+require 'rails_helper'
+require 'rake'
+
+RSpec.describe 'retention_instructions:backfill' do
+
+  shared_examples 'backfilling retention instructions' do
+
+    it 'backfills retention_instruction attribute in labware table' do
+      # Setup
+      labware = create(:labware, retention_instruction: nil,
+                                     custom_metadatum_collection:
+                                       create(:custom_metadatum_collection,
+                                              metadata: { 'retention_instruction' => 'Destroy after 2 years' }
+                                       )
+      )
+
+      # Execute
+      run_rake_task
+
+      # Verify
+      expect(labware.reload.retention_instruction).not_to be_nil
+    end
+
+    it 'removes existing retention instruction from custom_metadata table' do
+      # Setup
+      labware = create(:labware, retention_instruction: nil,
+                                     custom_metadatum_collection:
+                                       create(:custom_metadatum_collection,
+                                              metadata: { 'retention_instruction' => 'Destroy after 2 years' }
+                                       )
+      )
+
+      # Execute
+      run_rake_task
+
+      # Verify
+      expect(labware.reload.custom_metadatum_collection.metadata['retention_instruction']).to be_nil
+    end
+
+    it 'does not remove custom_metadatum_collection record for the labware' do
+      # Setup
+      labware = create(:labware, retention_instruction: nil,
+                                     custom_metadatum_collection:
+                                       create(:custom_metadatum_collection,
+                                              metadata: {
+                                                'retention_instruction' => 'Destroy after 2 years',
+                                                'other_key' => 'other_value'
+                                              }
+                                       )
+      )
+
+      # Execute
+      run_rake_task
+
+      # Verify
+      expect(labware.reload.custom_metadatum_collection).not_to be_nil
+    end
+
+    it 'removes only retention instructions from custom_metadata' do
+      # Setup
+      labware = create(:labware, retention_instruction: nil,
+                                     custom_metadatum_collection:
+                                       create(:custom_metadatum_collection,
+                                              metadata: {
+                                                'retention_instruction' => 'Destroy after 2 years',
+                                                'other_key' => 'other_value'
+                                              }
+                                       )
+      )
+
+      # Execute
+      run_rake_task
+
+      # Verify
+      expect(labware.reload.custom_metadatum_collection.metadata['other_key']).to eq('other_value')
```

Comment:
This one and the above are very similar, is the above one needed if this one checks the key exists, which if true means it cant be nil.

Reply:
So this test intends to check whether the script **does not** remove other metadata (i.e., it checks whether the script removes only the value with the key `retention_instruction` in `custom_metadata` table).

Reply:
Again similar to the above, probably different test preferences. If you are testing it does not remove other metadata is that not also testing that metadata is not nil?

Reply:
Yeah I guess you're correct. We could move this assertion into the test above, if that's what you're suggesting? 

I guess I've been following this "one-assertion per test" philosophy, which is [quite old](https://stackoverflow.blog/2022/11/03/multiple-assertions-per-test-are-fine/#h2-459b27e85d780). I need to get out of that one :D

Reply:
Yeah but I see your approach now and both are fine, happy to let you decide.